### PR TITLE
[Fabric] Fixes for 0.71 upgrade with Fabric enabled

### PR DESCRIPTION
**Pull request by shwanton**
## macOS - Fabric
CleanShot 2023-03-03 at 13 24 39

## iOS - Fabric
CleanShot 2023-03-03 at 13 44 43

### CODE REVIEW

File: React/Base/RCTUIKit.h [hunk 1]
```diff
@@ -369,7 +372,7 @@ CGPathRef UIBezierPathCreateCGPathRef(UIBezierPath *path);
 
 #define RCTPlatformWindow NSWindow
 
-@interface RCTUIView : NSView
+@interface RCTUIView : RCTPlatformView
```
Comment (Saadnajmi) [inline React/Base/RCTUIKit.h:375]: Curious, what break did this fix? Or was it more "why not?"

File: React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm [hunk 2]
```diff
@@ -33,7 +33,7 @@ @implementation RCTTextInputComponentView {
 #if !TARGET_OS_OSX // [macOS]
   RCTUIView<RCTBackedTextInputViewProtocol> *_backedTextInputView;
 #else // [macOS
-  RCTUITextView<RCTBackedTextInputViewProtocol> *_backedTextInputView;
+  RCTPlatformView<RCTBackedTextInputViewProtocol> *_backedTextInputView;
```
Comment (Saadnajmi) [inline React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:36]: Also this one?

**Review by Saadnajmi — APPROVED** [review]

File: React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm (hunk 2 above)
Comment (shwanton) [reply React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm:36]: `contentView` needs to be a `NSView` (aka `RCTPlatformView`) and not a RCTUIView. Most of our native controls do not inherit from `RCTUIView`

File: React/Base/RCTUIKit.h (hunk 1 above)
Comment (shwanton) [reply React/Base/RCTUIKit.h:375]: It seemed cleaner to me to not have multiple refs to `NSView`

**Pull request merged**